frontend_csv/src/parser/parser.rs

```diff
@@ -0,0 +1,230 @@
+//! Core Parser implementation for the csv tokens
+use std::collections::BTreeMap;
+use std::vec::Vec;
+
+use crate::parser::ast::LNMsData;
+use crate::parser::ast::LNMsg;
+use crate::parser::ast::LNTlvData;
+use crate::parser::ast::LNTlvType;
+use crate::scanner::token::{CSVToken, CSVTokenType};
+
+pub struct Parser {
+    pub simple_table: BTreeMap<String, LNMsg>,
+    pos: usize,
+    lnmsg_buffer: Option<LNMsg>,
+    lntlv_buffer: Option<LNTlvType>,
+}
+
+impl Parser {
+    /// Build a new parser
+    pub fn new() -> Self {
+        return Parser {
+            pos: 0,
+            simple_table: BTreeMap::new(),
+            lnmsg_buffer: None,
+            lntlv_buffer: None,
+        };
+    }
+
+    /// Take the element in the current position of the stream
+    fn peek<'a>(&self, tokens: &'a Vec<CSVToken>) -> &'a CSVToken {
+        return &tokens[self.pos];
+    }
+
+    /// Take the element in the current position of the stream
+    /// and increase the position by one
+    fn pop(&mut self, tokens: &Vec<CSVToken>) -> CSVToken {
+        let res = tokens[self.pos].clone();
+        self.pos += 1;
+        return res;
+    }
+
+    /// Parse a message type line of the csv file, where the format looks like
+    /// the following one:
+    ///
+    /// `msgtype,init,16`
+    fn parse_msg_typ(&mut self, tokens: &Vec<CSVToken>) {
+        let msg_name = self.pop(&tokens);
+        let msg_type = self.pop(&tokens);
+        match msg_type.ty {
+            CSVTokenType::Number => {
+                self.lnmsg_buffer = Some(LNMsg {
+                    msg_typ: msg_type.val.parse::<u64>().unwrap(),
+                    msg_name: msg_name.val.to_string(),
+                    msg_data: Vec::new(),
+                    tlv_stream: Vec::new(),
+                });
+            }
+            _ => panic!("Unknown Token {:?}", self.peek(&tokens)),
+        }
+    }
+
+    fn get_byte_length(&self, keyword: String) -> u64 {
```

Comment:
for what you use this?

Reply:
For getting the bitfield length, i add an example in the comment